rocksdb/columnfamily.nim

```diff
@@ -19,43 +19,48 @@
 
 {.push raises: [].}
 
-import
-  ./rocksdb
+import ./columnfamily/cfhandle, ./rocksdb
 
 export rocksdb
 
 type
   ColFamilyReadOnly* = object
     db: RocksDbReadOnlyRef
-    name: string
+    cfHandle: ColFamilyHandleRef
 
   ColFamilyReadWrite* = object
     db: RocksDbReadWriteRef
-    name: string
+    cfHandle: ColFamilyHandleRef
 
 proc withColFamily*(
-    db: RocksDbReadOnlyRef,
-    name: string): RocksDBResult[ColFamilyReadOnly] =
+    db: RocksDbReadOnlyRef, name: string
+): RocksDBResult[ColFamilyReadOnly] =
   ## Creates a new `ColFamilyReadOnly` from the given `RocksDbReadOnlyRef` and
   ## column family name.
+  doAssert not db.isClosed()
 
   # validate that the column family exists
-  discard db.keyExists(@[0.byte], name).valueOr:
+  let cfHandle = ?db.getColFamilyHandle(name)
+
+  discard db.keyExists(@[0.byte], cfHandle).valueOr:
```

Comment:
is this key-exists still necessary? shouldn't `getColFamilyHandle` already be performing the equivalent of this check?

Reply:
Good point, yes I believe it can be removed, will do.